### ISSUE TITLE
Explosion of library names and paths in subproject dependencies

### ISSUE THREAD

**Issue by nirbheek** [mesonbuild/meson#671]
If you have a sufficiently deeply-nested `declare_dependency` hierarchy with the `dependencies :` kwarg, you have the same library names and paths repeated over and over and over resulting in an explosive command line argument list filled with redundant paths and library names.

For example, the following code:
`library(...,   dependencies : [gstbase_dep, gstpbutils_dep, gstriff_dep, gstaudio_dep, gsttag_dep])`

Where each of those dependencies are declared as:

`declare_dependency(..., dependencies : [gst_base_dep, gst_dep, ...],... )`

i.e., they refer to each other, and to a bunch of common dependencies. This yields:

`cc    -o gst/wavparse/libgstwavparse.so 'gst/wavparse/gstwavparse@sha/gstwavparse.c.o' '-Wl,--no-undefined' '-Wl,--as-needed' '-shared' '-fPIC' '-Wl,-soname,libgstwavparse.so' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/pbutils/libgstpbutils-1.0.so.0.901.0' 'subprojects/gst-plugins-base/gst-libs/gst/video/libgstvideo-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/audio/libgstaudio-1.0.so.0.901.0' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/video/libgstvideo-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/audio/libgstaudio-1.0.so.0.901.0' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/riff/libgstriff-1.0.so.0.901.0' 'subprojects/gst-plugins-base/gst-libs/gst/audio/libgstaudio-1.0.so.0.901.0' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/audio/libgstaudio-1.0.so.0.901.0' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/audio/libgstaudio-1.0.so.0.901.0' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gst-plugins-base/gst-libs/gst/tag/libgsttag-1.0.so.0.901.0' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/libs/gst/base/libgstbase-1.0.so.0.901.0' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' 'subprojects/gstreamer/gst/libgstreamer-1.0.so.0.901.0' 'subprojects/gstreamer/gst/printf/libgstprintf.a' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lglib-2.0' '-lgobject-2.0' '-lglib-2.0' '-Wl,--export-dynamic' '-lgmodule-2.0' '-pthread' '-lglib-2.0' '-lm' 
`

And this is ridiculous. We should be de-duping this list of libraries while maintaining the order.

**Comment by jpakkane** [comment]
Dedupping `-I` and `-L` is something I have been thinking for a while and can be done without any side effects. Because library symbol resolution order is pants on head stupid, we can't just dedup blindly. If it were up to me (and also supported by VS?) we would just gather up all libs and pass them as a single clump with begin group and end group.

But could we do some dedupping smarter. Suppose for example if we have an external dep `foo` that is used in an internal dep `bar` as well as a target `fizz` that uses both `foo` and `bar`. Could we put the extdep there only once at the end? Would something break if it is there only once? (assuming shared libraries, with static it will definitely be possible to break)?

**Comment by tp-m** [comment]
Ignoring the main issue, I also wonder why the libgstprintf.a is repeated here everywhere, that should be private to libgstreamer-1.0.so.* and part of libgstreamer-1.0.so* and not appear in the list here imho.

**Comment by jpakkane** [comment]
It should appear for gstreamer.a but probably not for gstreamer.so

**Comment by nirbheek** [comment]
This is getting too close for comfort in gstreamer, with ninja failing to spawn executables when you use too many repeated recursive dependencies:

```
<thiblahute> got 'ninja: fatal: posix_spawn: Argument list too long' -- interesting one
```

**Comment by nirbheek** [comment]
This is fixed with #1545.
